packages/synapse-react-client/test/lib/containers/CreateProjectModal.test.tsx

```diff
@@ -34,28 +43,67 @@ describe('CreateProjectModal tests', () => {
   })
 
   it('Creates project on submit', async () => {
-    renderComponent()
-    const input: HTMLInputElement = (await screen.findByTestId(
-      'projectInput',
-    )) as HTMLInputElement
-    fireEvent.change(input, { target: { value: MOCK_PROJECT_NAME } })
+    const { user, input, saveButton } = setUp()
+
+    await user.type(input, MOCK_PROJECT_NAME)
     expect(input.value).toBe(MOCK_PROJECT_NAME)
-    fireEvent.keyDown(input, { key: 'Enter' })
+    await user.click(saveButton)
+
+    // should show success alert
+    await screen.findByText('Project created')
+  })
+
+  it('Creates project on enter', async () => {
+    const { user, input, saveButton } = setUp()
+
+    await user.type(input, MOCK_PROJECT_NAME)
+    expect(input.value).toBe(MOCK_PROJECT_NAME)
+    await user.keyboard('{ENTER}')
 
     // should show success alert
     await screen.findByText('Project created')
   })
 
   it('Shows error if creation fails', async () => {
-    renderComponent()
-    const input: HTMLInputElement = (await screen.findByTestId(
-      'projectInput',
-    )) as HTMLInputElement
-    fireEvent.change(input, { target: { value: MOCK_INVALID_PROJECT_NAME } })
+    const { user, input, saveButton } = setUp()
+
+    await user.type(input, MOCK_INVALID_PROJECT_NAME)
     expect(input.value).toBe(MOCK_INVALID_PROJECT_NAME)
-    fireEvent.keyDown(input, { key: 'Enter' })
+    await user.click(saveButton)
 
     // should show error alert
     await screen.findByText('Invalid project name')
   })
+
+  it('Clears alert and project name on cancel', async () => {
+    const { user, input, saveButton } = setUp()
+
+    await user.type(input, MOCK_INVALID_PROJECT_NAME)
+    await user.click(saveButton)
+
+    const alert = await screen.findByRole('alert')
+    expect(alert).toHaveTextContent('Invalid project name')
+
+    const cancelButton = screen.getByRole('button', { name: /cancel/i })
+    await user.click(cancelButton)
+
+    expect(alert).not.toBeInTheDocument()
+    expect(input.value).toBe('')
+  })
+
+  it('Clears alert and project name on closing dialog', async () => {
+    const { user, input, saveButton } = setUp()
+
+    await user.type(input, MOCK_INVALID_PROJECT_NAME)
+    await user.click(saveButton)
+
+    const alert = await screen.findByRole('alert')
+    expect(alert).toHaveTextContent('Invalid project name')
+
+    const closeButton = screen.getByRole('button', { name: /close/i })
+    await user.click(closeButton)
+
+    expect(alert).not.toBeInTheDocument()
+    expect(input.value).toBe('')
+  })
```

Comment:
Test that clicking "Cancel" button or clicking modal close button result in the same side effects